libbeat/outputs/elasticsearch/client.go

```diff
@@ -245,7 +247,11 @@ func (client *Client) publishEvents(ctx context.Context, data []publisher.Event)
 		return nil, nil
 	}
 
-	status, result, sendErr := client.conn.Bulk(ctx, "", "", nil, bulkItems)
+	params := map[string]string{}
+	if client.conn.BulkResponseFiltering {
```

Comment:
Do we need this conditional? Why not always set the `filter_path` param here?

Reply:
discussed this among others over Zoom, the use of the config flag is just for having an escape hatch if this change is causing issues.
While moving the code one level up to the output package (after commit 810c14071b66e39691acbe6e7dbf6ae181eec994) I noticed that all of the tests we currently have on the client are already mock-testing with a filtered response and not the full response, example https://github.com/elastic/beats/blob/d7403fd5e002d7b5ab67caa26b4719545d1de828/libbeat/outputs/elasticsearch/client_test.go#L406 

Given the above, It might not make sense to even add that flag.
With the help of @ycombinator to help me review this PR we will decide if we do actually need it or not.

Reply:
Yep, given that we have a unit test for `client.bulkCollectPublishFails()` that only uses the filtered response anyway, and that test has been passing even before the `filter_path` improvement in this PR, we can be confident that the behavior of `client.bulkCollectPublishFails()` will be unaffected by the change in this PR.  Further, I checked the code and it appears that `client.bulkCollectPublishFails()` is the only method that consumes the response from the ES bulk API. 

We should remove the conditional here and the associated `BulkResponseFiltering` field from the `eslegclient.ConnectionSettings` and `elasticsearch.elasticsearchConfig` structs.

Reply:
thanks for the verification, I removed it in 2b1bee29de3a822cf1c615e1b4681811b94e4afa